python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -354,7 +354,8 @@ def as_device_scalar(val, dtype=None):
 def _is_null_host_scalar(slr):
     if cudf.utils.utils.is_na_like(slr):
         return True
-    elif isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr):
+    elif (isinstance(slr, (np.datetime64, np.timedelta64)) and np.isnat(slr)) or \
+            isinstance(slr, type(pd.NaT)):
```

Comment:
Nit: Could you use `slr is pd.NaT` (and below)?

Reply:
Done